extensions/azurecore/src/account-provider/auths/azureAuth.ts

```diff
@@ -1107,4 +1114,12 @@ export interface DeviceCodeCheckPostData extends Omit<TokenPostData, 'resource'>
 	tenant: string,
 	code: string
 }
+
+export interface ResponseData<T> {
+	error?: {
```

Comment:
This probably isn't really how it should be defined (error should be its own type) - but a quick glance through the ms-rest typings we have currently didn't show anything about the exact typings for the responses and the docs only really cover the success scenarios. 

So this will currently solve the issue I was initially trying to fix with the compile time checking, but know that this is all based on assuming that the responses are all of the same type - and given that we use this in multiple places (including ADAL) that seems unlikely to be the case. Not going to spend much time on this though if you're going to be replacing this in the nearish future.

Reply:
By "let it stay" you mean not make this change? Are you going to be deleting this function then?